### ISSUE TITLE
New concept for entering recurrence rules / full-featured recurrence settings

### ISSUE THREAD

**Issue by georgehrke** [owncloud/calendar#223]
The recurrence editor in the sidebar is currently being implemented.
It will follow the example of other calendar applications.

Although the current concept is pretty wide-spread, it's not really intuitive and we should create a concept for something more user-friendly.

This is definitely something to discuss at the next hackathon or conference.

**Comment by DemoFreak** [comment]
Does the "Maybe some day" milestone means that in OC9.0.0 it isn't possible to create/edit recurring events? ;-)

**Comment by georgehrke** [comment]
The description says: 

 > The recurrence editor in the sidebar is currently being implemented.

So you will be able to enter recurrence rules with ownCloud 9, but like in most other calendaring applications, this is not very user friendly. this issue is about creating a whole new concept for entering recurrence rules.

**Comment by DemoFreak** [comment]
Fine, sorry for bothering.
"Is currently being implemented" in conjunction with "maybe some day" was a bit ambiguous to me, my mistake. ;-)

**Comment by georgehrke** [comment]
@owncloud/designers 

You can get a rough idea what different options are available in https://tools.ietf.org/html/rfc5545#section-3.8.5.3

**Comment by georgehrke** [comment]
We should aim to support all possible rules but FREQ=SECONDLY,MINUTELY and HOURLY.

The repeating events editor in the rework will only support the most common options for now.

**Comment by georgehrke** [comment]
- [ ] automatically update WKST when a user touches the RRULE or offer some option?

**Comment by nksupport** [comment]
Hi Georg! I moved from Google Drive to OwnCloud 8 last year and the last part to be done is the calendar - been waiting for implementation of https://github.com/owncloud/apps/issues/23 and https://github.com/owncloudarchive/calendar/issues/1005. It's quite a useful feature in my workflow and importing recurring events from Google still doesn't work as of today. If you modified one event out of the recurring series in google, they're all just broken upon import (dates are a mess). I didn't necropost the old thread, but please know that there still are people waiting and hoping for the new design that works. Just updated to 9.0.0 beta2 and cloned the rework yesterday. Steps to reproduce: create a recurring event series in google, move/edit one of them, export to .ics and import into owncloud:
1. moving one of the events in the series changes start/end dates on all of them, that's the known issue of the old system.
2. import of such events is broken in its own way: the event from [broken.txt](https://github.com/owncloud/calendar/files/158572/broken.txt) attached, with the start/end dates both on 2015-09-16, is placed on the grid by the new calendar on 2015-12-03 for reasons i don't quite grasp. Editor shows the correct dates.  It's a part of a monthly recurring series, and it has been modified directly. It's off by a good two and half month. 
3. None of the future recurrences made it through the import, only the past ones are shown, and those are shown at the wrong dates.

> The recurrence editor in the sidebar is currently being implemented.
> It will follow the example of other calendar applications.

Please draw inspiration from google calendar's interface for recurring events, it's the most flexible and intuitive solution i've ever seen. Every option you look for is always where you expect it. 

P.S. The .ics import dialog offers the option to create a new calendar but doesn't offer the color picker, and i found no editor to change colors once the calendar is imported. Probably something you know of, so i didn't create a new issue.

**Comment by georgehrke** [comment]
@jancborchardt We also need to talk about this issue on the conf and create some concept/mockups

**Comment by georgehrke** [comment]
For reference: https://gist.github.com/georgehrke/182bb08b75afc4d5a729e9d552a19316
(previous work on the full-featured recurrence editor)

**Comment by jancborchardt** [comment]
@georgehrke yeah, let’s talk about it at the conf! :) Did you already submit a talk about the developments on the Calendar app? https://conf.nextcloud.com/conference/NextcloudConference2016/program/proposal/new

**Comment by scriptPilot** [comment]
Hey, what is the current situation to this topic? 

To set "until" dates (#484) my workflow so far is:

1) Export ownCloud Calendar
2) Import in Google Calendar
3) Apply changes
4) Export Google Calendar
5) Delete ownCloud Calendar
6) Import Google Calendar to ownCloud

**Comment by jancborchardt** [comment]
@scriptPilot as said above, @georgehrke and I will chat about this at the [Nextcloud conference](https://conf.nextcloud.com). :)

**Comment by staehler** [comment]
Hi,
not sure if you talk about the same problem, I have. I synchronize my outlook calendar (Outlook CalDav Synchronizer) with owncloud 9.1.3 and my recurring events mess up in owncloud. Deleted items in outlook remain in owncloud and moved events in outlook duplicate the entries in owncloud (the old event remains and a new one is created). 
When I check the recurrency rule on such an event I get the following message (german):

> Die Wiederholungsregel dieses Termins wird noch nicht unterstützt. Unterstützung für fortgeschrittene Regeln wird mit späteren Aktualisierungen hinzugefügt.

Is this the bug in owncloud (discussed above) or a bug in Outlook CalDav Synchronizer?

**Comment by strugee** [comment]
@staehler not sure, but it's not a bug in this repo. The underlying CalDAV support is in oC/NC core.

**Comment by t7ko** [comment]
From what I read above, I'm not sure: does this proposal include support for "repeat every weekday" option?  Or an ability to select specific days of the week for weekly repeat?
